Add the person, student and teacher classes

medsonmoombe opened the pull request:
### In this PR, I did the following:

- Create class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Create class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Create class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Overrode `can_use_services?` so it always returns `true`.
- Each class was saved in a separate file.

medsonmoombe committed:
Add the person, student and teacher classes

medsonmoombe committed:
fix linter errors

medsonmoombe committed:
update README.md file and fix linter errors

TedLivist commented on student.rb:
```diff
@@ -0,0 +1,12 @@
+
+class Student < Person
+  def initialize(classroom, age, name: 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
- [ ] I tried creating a student and got an error relating to the arguments expected. Since your `name` and `parent_permission` attributes are defined using the colon notation, you need to also assign your arguments in the `#super` method, like so:

  ```
  super(age, name: name, parent_permission: parent_permission)
  ```

TedLivist commented on teacher.rb:
```diff
@@ -0,0 +1,14 @@
+  attr_reader :specialization
+
+  def initialize(specialization, age, name: 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
- [ ] Kindly re-assign your `#super` parameters as demonstrated above.

TedLivist commented on person.rb:
```diff
@@ -0,0 +1,21 @@
+  end
+
+  def can_use_services?
+    of_age || @parent_permission == true
```
- [ ] I believe this should be `of_age?`, (with the question mark)

- [ ] `@parent_permission == true` is unnecessary. By definition, `@parent_permission` is either `true` or `false`, so we can remove the equality sign
  
  We should have something like so:
  ```
  def can_use_services?
    of_age? || @parent_permission
  end
  ```

TedLivist reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @medsonmoombe,

Good job so far!
✔️ Good documentation
✔️ Imports of custom classes are made using `require_relative` according to convention
✔️ Classes are in separate files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

medsonmoombe committed:
Updated the class and methods

medsonmoombe commented:
### The following were implemented as requested  changes by the reviewer
- updated the person class to be able to add new class
- corrected the error method of_age?
- And updated the conditional for the parent_permission

DammyShittu reviewed (approved):
Hi @medsonmoombe,

## APPROVED 🟢

### Highlights
✔️ Descriptive PR
✔️ Private method

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
